### ISSUE TITLE
no-import-private: a module file named `private.ts` is reported as a private directory

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#607]
### Describe the bug

`no-import-private` decides from the import path string alone. When a **module file** is named `private.ts`, an import of it from another directory resolves to a path whose last segment is `private`, and the rule treats that segment as a `private` directory and reports the import.

```
src/
  c/
    private.ts      export const helper = 1;
  d/
    use.ts          import { helper } from "../c/private";
```

Rule enabled: `awscdk/no-import-private: "error"`.

```
$ npx eslint src/d/use.ts
  1:1  error  Cannot import modules from private dir at different levels of the hierarchy  awscdk/no-import-private

$ npx oxlint --format=unix -c .oxlintrc.json src/d/use.ts
src/d/use.ts:1:1: Cannot import modules from private dir at different levels of the hierarchy. [Error/awscdk(no-import-private)]
```

### Expected Behavior

No report: `../c/private` is a regular module (`c/private.ts`), not a `private/` directory. Imports of a real `private/` directory (`../c/private/x`, `../c/private` resolving to `c/private/index.ts`) keep being reported.

### Current Behavior

Both plugins report the import of `c/private.ts`.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.2 / oxlint-plugin-awscdk 1.0.2

### Language Version

TypeScript 6.0.3 (eslint) / TypeScript 7.0.2 via oxlint-tsgolint 7.0.2001 (oxlint)

### Additional Context

```yaml
# machine-readable summary
rules: ["no-import-private"]
symptom: ["false-positive"]
affected_packages: ["eslint-plugin-awscdk", "oxlint-plugin-awscdk"]
root_cause: "packages/*/src/rules/no-import-private.ts:65-69 (the segments of the resolved import path include the final module segment, so lastIndexOf(\"private\") also matches a module named private)"
verified_on: "main @ 9caea55 (eslint-plugin-awscdk 4.4.2, oxlint-plugin-awscdk 1.0.2)"
related_issues: ["#544"]
```

Verified fix direction (eslint): when `private` is the **last** segment of the resolved import path and a file `<path>.ts` / `.tsx` / `.mts` / `.cts` / `.js` / `.mjs` / `.cjs` / `.d.ts` exists, treat the specifier as a module import and return. Only the module-file case stops being reported; directory imports (`./private`, `../x/private`, `../x/private/y`) are unchanged, the existing unit tests (virtual file names, nothing on disk) still pass, and `examples/` stays at 43 errors. A specifier with an explicit extension (`../c/private.js`) is not affected today because its last segment is `private.js`.

Reproduction environment:

- macOS 14.1, Node.js v24.18.0, pnpm 10.29.3
- aws-cdk-lib 2.268.0 / constructs 10.8.1
- ESLint 10.9.1 + typescript-eslint 8.69.0 + TypeScript 6.0.3
- oxlint 1.81.0 + oxlint-tsgolint 7.0.2001 (corsa-oxlint 1.13.1)
